src/table/src/TableVirtualBody.js

```diff
@@ -1,136 +1,84 @@
-import React, { PureComponent } from 'react'
+import React, { memo, useState, useEffect } from 'react'
 import PropTypes from 'prop-types'
 import VirtualList from 'react-tiny-virtual-list'
 import debounce from 'lodash.debounce'
 import { Pane } from '../../layers'
 
-export default class TableVirtualBody extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes the Pane component as the base.
-     */
-    ...Pane.propTypes,
-
-    /**
-     * Children needs to be an array of a single node.
-     */
-    children: PropTypes.oneOfType([
-      PropTypes.arrayOf(PropTypes.node),
-      PropTypes.node
-    ]),
-
-    /**
-     * Default height of each row.
-     * 48 is the default height of a TableRow.
-     */
-    defaultHeight: PropTypes.number,
-
-    /**
-     * When true, support `height="auto"` on children being rendered.
-     * This is somewhat of an expirmental feature.
-     */
-    allowAutoHeight: PropTypes.bool,
-
-    /**
-     * The overscanCount property passed to react-tiny-virtual-list.
-     */
-    overscanCount: PropTypes.number.isRequired,
-
-    /**
-     * When passed, this is used as the `estimatedItemSize` in react-tiny-virtual-list.
-     * Only when `allowAutoHeight` and`useAverageAutoHeightEstimation` are false.
-     */
-    estimatedItemSize: PropTypes.number,
-
-    /**
-     * When allowAutoHeight is true and this prop is true, the estimated height
-     * will be computed based on the average height of auto height rows.
-     */
-    useAverageAutoHeightEstimation: PropTypes.bool,
-
-    /**
-     * The scrollToIndex property passed to react-tiny-virtual-list
-     */
-    scrollToIndex: PropTypes.number,
-    /**
-     * The scrollOffset property passed to react-tiny-virtual-list
-     */
-    scrollOffset: PropTypes.number,
-    /**
-     * The scrollToAlignment property passed to react-tiny-virtual-list
-     */
-    scrollToAlignment: PropTypes.oneOf(['start', 'center', 'end', 'auto']),
-    /**
-     * The onScroll callback passed to react-tiny-virtual-list
-     */
-    onScroll: PropTypes.func
+const useForceUpdate = () => useState()[1]
```

Comment:
I dont understand this

Reply:
Are you trying something like this? https://stackoverflow.com/questions/53215285/how-can-i-force-component-to-re-render-with-hooks-in-react

Reply:
essentially yes, just trying to force a rerender

Reply:
I'm not sure what you have will actually work. What you want is something like this:

```js
const useForceUpdate = () => {
  const [_, setValue] = useState()

  return useCallback(() => {
    setValue({}) // set with a new object instance every time it gets called
  }, [setValue])
}
```

Reply:
It does seem to work, but if you think that what you have there would work out better, I can update it to that

Reply:
I think you need to call set state with a new object for it to rerender